Cython/Compiler/Parsing.py

```diff
@@ -3002,9 +3002,15 @@ def p_with_gil(s):
         return 0
 
 def p_exception_value_clause(s):
+    # the default behaviour is to always
+    # check for exceptions
+    exc_check = 1
     exc_val = None
-    exc_check = 0
-    if s.sy == 'except':
+    if s.sy == 'IDENT' and s.systring == 'noexcept':
+        s.next()
+        exc_val = None
+        exc_check = 0
+    elif s.sy == 'except':
         s.next()
         if s.sy == '*':
             exc_check = 1
```

Comment:
This line can now be removed (similar below).

In contrast, the `except -1` case (i.e. the as-of-now still non-existent else-branch of `if s.sy == '?'`) will need an `exc_check = 0`.

Reply:
Could you please help me understand what it means when `exc_val != None` but `exc_check = 0`?

Reply:
Aside from just keeping parity with the way the values are set currently (my starting point for the comments I made), you can check out the difference in the [docs](https://docs.cython.org/en/latest/src/userguide/language_basics.html#error-return-values) between `except? -1` (checked)  and `except -1` (unchecked).

Reply:
Ah thanks -- I was misunderstanding what `exc_check = False` implied. I thought it meant no check is done at all.

It looks like if `exc_val != None`, then a check is done if the return value is `exc_val`, even if `exc_check = 0`.